---
title: "Book notes: Bayesian Evolutionary Analysis with BEAST (theory)"
date: 2024-12-18
permalink: /posts/2024-12-18/BEAST2-book
categories:
  - Book notes
tags:
  - Phylogenetics
  - BEAST2
toc: true
last_modified_at: 2024-12-26
---

This is the BEAST2 book by Alexei J. Drummond and Remco R. Bouckaert. I will focus on the Part I of this book, which is the theory part. Part III also seems to be interesting, it is on programming.

Below are some notes.

# Introduction

## Molecular phylogenetics

- The statistical treatment of phylogenetics was made feasible by Felsenstein (1981), who described a computationally tractable approach to computing the probability of the sequence alignment given a phylogenetic tree and a model of molecular evolution, $Pr\{D\vert T,\Omega\}$. This quantity is known as the *phylogenetic likelihood* of the tree and can be efficiently computed by the peeling algorithm.
- A continuous-time Markov process (CTMP) can be used to describe the evolution of a single nucleotide, assuming iid of the sites, this can be extended to multiple sequence alignment.
- Bayesian inference have great flexibility in describing prior knowledge, and it can apply to complex highly parametric models via MCMC.

## Coalescent theory

- A gene tree is estimated from individuals **sampled** from a population, the size of the population can be estimated using Kingman's coalescent model.
- *Genealogy-based population genetics* methods can infer many parameters governing molecular evolution and population dynamics, including effective population size, rate of population growth/decline, migration rates, population structure, recombination rates and reticulate ancestry.
- [Kingman's coalescent](https://alexeidrummond.org/bayesian_phylo_lectures/lecture10/?print-pdf#/).
- *Incomplete lineage sorting* (ILS) is commonly modeled using the multispecies coalescent framework, which integrates a population-level coalescent process (gene tree) into a phylogenetic (species) tree.

## Virus evolution and pholodynamics

- Because RNA viruses evolve so quickly, the “evolutionary clock” (the genetic changes in virus populations) runs at a similar pace to the “ecological clock” (the spread and decline of infections, changes in host populations, and alterations in transmission routes).
- Neutral genetic variation—mutations not heavily influenced by natural selection—provides a molecular record of the virus’s past.

## Before and beyond tress

- Sequence alignment:
  - ClustalW and ClustalX uses a guide tree constructed by a distance-based method to progressively construct MSA via pairwise alignment. This guide tree may be a bias if you aim to reconstruct the phylogeny using the MSA.
  - T-coffee builds a library of pairwise alignments to construct a MSA.
  - MUSCLE and MAFFT are suitable for high-throughput MSA.
  - *Statistical alignment* reconstructs the MSA and the phylogeny simultaneously, using a probabilistic model of sequence evolution.
- Ancestral recombination graph (ARG) is a generalization of a genealogy that includes recombination events.

## Probability and Bayesian inference

- Bayes' formula: $P(\theta\vert D) = \frac{P(D\vert \theta)P(\theta)}{P(D)}$.
- In a phylogenetic context, $f(\theta\vert D) = \frac{Pr(D\vert\theta)f(\theta)}{Pr(D)}$, where D is the sequence data, $\theta$ is the set of parameters, including the tree, substitution model parameters, clock rates, etc. $f(\theta\vert D)$ is the *posterior distribution*, $Pr(D\vert \theta)$ is the *likelihood*, $f(\theta)$ is the *prior distribution*, and $Pr(D)$ is the *marginal likelihood*.
- The marginal likelihood is the probability of the data averaged over all possible values of the parameters, $Pr(D) = \int Pr(D\vert \theta)f(\theta)d\theta$. It is almost always intractable to compute, but it can be estimated using MCMC.
- Non-informative priors hold a natural appeal, but they can be *improper* in a formal sense if they do not integrate to 1 over the parameter space. Such improper priors are extremely dangerous in a Bayesian analysis, as they can lead to improper posteriors.
- In MCMC, the new state is accepted with probability $\alpha = min\{1, \frac{f(\theta'\vert D)}{f(\theta\vert D)}\}$, where $\theta'$ is the proposed new state. This algorithm assumes that the proposal distribution is symmetric, where $q(\theta'\vert \theta) = q(\theta\vert \theta')$.
- Effective sample size (ESS) is a measure of the number of independent samples in a chain, accounting for autocorrelation. 
- The Metropolis-Hastings algorithm is a generalization of the Metropolis algorithm that allows for asymmetric proposal distributions. MCMC maintains reversibility by factoring in a Hastings ratio, $\alpha = min\{1, \frac{f(\theta'\vert D)q(\theta\vert \theta')}{f(\theta\vert D)q(\theta'\vert \theta)}\}$.
- Gibbs sampler is a special case of the Metropolis-Hastings algorithm, where the proposal distribution is the conditional distribution of each parameter given the others.
- When the parameter space is not of a fixed dimension, reversible-jump MCMC or Bayesian variable selection (BSVS) can be used to explore the space of models with different numbers of parameters.
- Bayes factors are used to compare the fit of different models to the data, $BF = \frac{Pr(D\vert M_1)}{Pr(D\vert M_2)}$, which rely on the marginal likelihoods of the models. 
-  Path sampling is a computational technique used in Bayesian phylogenetic inference to estimate the marginal likelihood by slowly “turning on” the likelihood function. By evaluating how the expected log-likelihood changes as you go from the prior distribution ($β=0$) to the posterior distribution ($β=1$), you obtain a numerical approximation of the marginal likelihood—an essential quantity for rigorous Bayesian model comparison.
-  Maximum likelihood methods find the $max_\theta Pr(D\vert \theta)$. Bayesian inference finds the posterior distribution $f(\theta\vert D)$. 
- The posterior distribution lets you reason about how plausible different parameters are, not just the single best estimate.

# Evolutionary trees

## Types of trees

- A rooted binary tree of $n$ leaves has $2n-1$ nodes (both internal nodes and tips) and $2n-2$ branches.
- Multifurcating trees (as opposed to bifurcating/binary trees) are those that have one or more polytomies.
- In time trees,  the times of the internal nodes are called divergence times, ages or coalescent times, while the times of the leaves are known as tip times or sampling times.
- When inferring rooted tress, we link the amount and duration of evolution. If amount and duration is the same, we assume a strict clock model. If there is large variance between the two, we assume a relaxed clock model.
- Since in a time-tree node heights correspond to the ages of the nodes, such rooted tree models have fewer parameters than unrooted tree models, approaching roughly half for large number of taxa (for $n$ taxa, there are $n-1$ node heights for internal nodes and a few parameters for the clock model, but $2n-3$ internal nodes in an unrooted tree).

## Counting trees

- The number of tip-labelled rooted binary trees for $n$ taxa is:
  $$T_n = \prod_{k=2}^n {(2k-3)} = (2k-3)(2k-5)...(7)(5)(3)(1).$$

- The number of unlabelled rooted tree shapes, the number of labelled rooted
trees, the number of labelled ranked trees (on contemporaneous tips) and the number of fully ranked trees (on distinctly timed tips) as a function of the number of taxa, $n$

  | n   | #shapes | #trees ($T_n$) | #ranked trees ($R_n$) | #fully ranked trees ($F_n$) |
  |-----|---------|------------|-----------------------|---------------------------|
  | 2   | 1       | 1          | 1                     | 1                         |
  | 3   | 1       | 3          | 3                     | 4                         |
  | 4   | 2       | 15         | 18                    | 34                        |
  | 5   | 3       | 105        | 180                   | 496                       |
  | 6   | 6       | 945        | 2,700                 | 11,056                    |
  | 7   | 11      | 10,395     | 56,700                | 349,504                   |
  | 8   | 23      | 135,135    | 1,587,600             | 14,873,104                |
  | 9   | 46      | 2,027,025  | 57,153,600            | 819,786,496               |
  | 10  | 98      | 34,459,425 | 2,571,912,000         | 56,814,228,736            |

- The number of tree shapes (or unlabelled rooted tree topologies) of $ n $ taxa, $ a_n $, is given by (Cavalli-Sforza and Edwards 1967):

  $$
  a_n =
  \begin{cases} 
  \sum_{i=1}^{(n-1)/2} a_i a_{n-i}, & \text{if } n \text{ is odd} \\
  a_1 a_{n-1} + a_2 a_{n-2} + \dots + \frac{1}{2}a_{n/2}(a_{n/2} + 1), & \text{if } n \text{ is even}.
  \end{cases}
  $$

- Ranked tree differ from unranked trees in that the internal nodes are ordered, and fully ranked trees are those where the tips are ordered by time of sampling.

- There are more ranked trees than rooted tree topologies, and they are important because many natural tree priors are uniform on ranked trees rather than tree shapes. The number of ranked trees of $ n $ contemporaneous taxa, $F(n) = \vert R_n\vert $, is:

  $$
  F(n) = |R_n| = \prod_{k=2}^n \binom{k}{2} = \frac{n!(n-1)!}{2^{n-1}}.
  $$

- All ranked trees with four tips are shown in Figure 2.3. When a tree has non-contemporaneous times for the sampled taxa, we term the tree **fully ranked** (Gavryushkina et al. 2013), and the number of fully ranked trees of $ n $ tips can be computed by recursion:

  $$
  F(n_1, \dots, n_m) = \sum_{i=1}^{n_m} \frac{|R_{n_m}|}{|R_i|} F(n_1, n_2, \dots, n_{m-2}, n_{m-1} + i),
  $$

  where $ n_i $ is the number of tips in the $ i $-th set of tips, grouped by sample time (see Gavryushkina et al. 2013 for details).

  ![Figure 2.3](/files/2024-12-18-BEAST2-book/IMG_0015.jpg)

## The Coalescent

- The Wright-Fisher model in its simplest form assumes (1) constant population size $N$, (2) discrete generations, (3) complete mixing.

- The below example shows when two individuals sampled from the current generation traced back in time. The probability of coalescence at $t$ generations is $Pr(t)=\frac{1}{N}(1-\frac{1}{N})^{t-1}$. The expected waiting time for coalescence is $E(\tau_k)=\frac{N}{K\choose2}$, where $k$ is the number of lineages.
  ![Wright-Fisher Model Example](/files/2024-12-18-BEAST2-book/Screenshot%202024-12-20%20at%2017.01.16.png){ width=500pt }
- Kingman shows that as $N$ grows the coalescent process converges to a CTMC, where the waiting time for coalescence is exponentially distributed with rate $\frac{K\choose2}{N}$.
- Effective population size may be useful for comparing different populations, however see Gillespie (2001) for an argument that these neutral evolution models are irrelevant to much real data because neutral loci will frequently be sufficiently close to loci under selection, that genetic draft and genetic hitchhiking will destroy the relationship between population size and genetic diversity that coalescent theory relies on for its inferential power.

### Coalescent with changing population size in a well-mixed population

- Parametric models with a pre-defined population function, such as exponential growth, expansion model and logistic growth models can easily be used in a coalescent framework.
- Non-parametric models, such as skyline plot models treat the coalescent intervals as separate segments. The generalized skyline plot can group the intervals according to the small-sample Akaike information criterion (AICc). Generalized skyline plots can be implemented in a Bayesian framework, which simultaneously infers the sample genealogy, the substitution parameters and the population size history.
- The Bayesian skyline plot analysis of a data set collected from a pair of HIV-1 donor and recipient was used to reveal a substantial loss of genetic diversity following virus transmission (Edwards et al. 2006). A further parametric analysis assuming constant population size in the donor and logistic growth model in the recipient estimated that more than 99% of the genetic diversity of HIV-1 present in the donor is lost during horizontal transmission. 
- Serially sampled coalescent: The tree shape probability is conditioned on sampled $\tau$ time:
  ![](/files/2024-12-18-BEAST2-book/IMG_05944E6CDC89-1.jpeg)

### Modelling epidemic dynamics using coalescent theory

- SIR type models can be combined with coalescent theory to model the spread of infectious diseases. Currently, the probabilities of phylogenetic trees can only be solved analytically for small host population sizes in the simplest endemic setting (SI and SIS) (Leventhal et al. 2014). 
- Recall that the coalescent calculates the probability density of a tree given the coalescent rate. The coalescent rate for $k$ lineages is $k\choose2$ times the inverse of the product of effective population size $N_e$ and generation time $g_c$. Volz (2012) proposed a coalescent approximation to epidemiological models such as the SIR, where the effective population size $N_e$ is the expected number of infected individuals through time, and the generation time $g_c$.

## Birth-death models

- In coalescent models, coalescent times only depend on deterministic
population size, meaning population size is the only parameter in the coalescent. If the population size is stochastic and small, the coalescent model may not be appropriate. Birth-death models can be used to model the population size as a stochastic process.

### Constant-rate birth-death models

- The continuous-time constant-rate birth–death process is a birth–death process which starts with one lineage at time $z_0$ in the past and continues forward in time with a stochastic rate of birth ($λ$) and a stochastic rate of death ($μ$) until the present (time $0$). At present, each extant lineage is sampled with probability $ρ$. 
- The constant-rate birth–death model with $μ = 0$ and $ρ = 1$, i.e. no extinction and complete sampling, corresponds to the well-known Yule model (Edwards 1970; Yule 1924).
- The three parameters $λ, μ, ρ$ are non-identifiable, meaning that the probability density of a time-tree is determined by the two parameters $λ− μ$ and $λρ$ (the probability density only depends on these two parameters if the probability density is conditioned on survival or on $n$ samples (Stadler 2009)). Thus, if the priors for all three parameters are non-informative, we obtain large credible intervals when estimating these parameters
- One difference between the constant-rate birth–death process limit and the coalescent is that the birth–death process induces a stochastically varying population size while classic coalescent theory relies on a deterministic population size.

### Time-dependent birth-death models

- The piecewise constant birth–death process is analogue to the coalescent skyline
plot where the population size is piecewise constant, thus we call this model the birth–death skyline plot (Stadler et al. 2013). The birth–death skyline plot has been used, for example, to reject the hypothesis of increased mammalian diversification following the K/T boundary (Meredith et al. 2011; Stadler 2011).

### Serially sampled birth-death models

- Since in the previous section we only defined a sampling probability $ρ$ at present, the birth–death model only gives rise to time-trees with contemporaneous tips. Stadler (2010) extended the constant rate birth–death model to account for serial sampling by assuming a sampling rate $ψ$. This means that each lineage is sampled with a rate $ψ$ and it is assumed that the lineage goes extinct after being sampled.

## Trees within trees

### The multispecies coalescent

- So far we assumed that the genealogy equals the species or transmission tree. However, this is an approximation. The genealogy is actually embedded within the species/transmission tree.
- The multispecies coalescent model can be used to estimate the species time-tree $g_S$, together with ancestral population sizes $N$, given the sequence data from multiple genes, whose gene trees may differ due to incomplete lineage sorting (Pamilo and Nei 1988).

### Viral transmission histories

- Nested time-trees can be estimated from a single data set is in the case of estimating transmission histories from viral sequence data, where the nested time-tree is the viral gene tree, and the encompassing tree is the tree describing transmissions between hosts (Leitner and Fitch 1999). 
- The natural generative model at the level of the host population is a branching process of infections, where each branching event represents a transmission of the disease from one infected individual to the next, and the terminal branches of this transmission tree represent the transition from infectious to recovery or death of the infected host organism. For multicellular host species there is an additional process of proliferation of infected cells within the host’s body (often restricted to certain susceptible tissues) that also has a within-host branching process of cell-to-cell infections. This two-level hierarchical process can be extended to consider different infectious compartments at the host level, representing different stages in disease progression, and/or different classes of dynamic behaviour among hosts.
- Accepting the above as the basic schema for the generative process, one needs to also consider a typical observation process of an epidemic or endemic disease. It is often the case that data are obtained through time from some fraction, but not all, of the infected individuals. Figure 2.9 illustrates the relationship between the full transmission history and the various sampled histories. 
  ![](/files/2024-12-18-BEAST2-book/IMG_82EF4F09AC5C-1.jpeg)
- MASTER/REMASTER can model this schema (combining both the generative and observational processes).

# Substitution and site models

- Hamming distance tend to **underestimate** the number of substitutions, because it does not account for multiple substitutions at the same site.
- *Distance correction* requires an explicit model of substitution, such as the Jukes-Cantor model. Under JC69, an estimate for the evolutionary distance between two sequences is $d = -\frac{3}{4}ln(1-\frac{4}{3}p)$, where $p$ is the proportion of sites that differ between the two sequences. For $p=0.4$, $\hat{d} \approx 0.571605$. This is an estimate of the **expected number of substitutions per site**.

## Continuous-time Markov process

- Assuming that the Markov process is in state $i \in \{A, C, G, T\}$ at time $t$, then in the next small moment, the probability that the process transitions to state $j \in \{A, C, G, T\}$ is governed by the *instantaneous transition rate matrix* $Q$:
  $$
  Q =
  \begin{bmatrix}
  \cdot & q_{AC} & q_{AG} & q_{AT} \\
  q_{CA} & \cdot & q_{CG} & q_{CT} \\
  q_{GA} & q_{GC} & \cdot & q_{GT} \\
  q_{TA} & q_{TC} & q_{TG} & \cdot
  \end{bmatrix}
  $$

- The total flow out of nucleotide state $i$ and are thus negative. At equilibrium, the total rate of change per site per unit time is thus:
  $$
  \mu = -\sum_i \pi_i q_{ii},
  $$
  where $π_i$ is the probability of being in state i at equilibrium, so that $μ$ is just the weighted
average outflow rate at equilibrium.

- The *transition probability matrix*, $P(t)$, provides the probability for being in state $j$ after time $t$, assuming the process began in state $i$:
  $$
  P(t) = \exp(Qt).
  $$

## DNA models

- All transition probabilities will asymptote equilibrium base frequency (e.g., $\frac{1}{4}$ if base frequencies are equal) for very large $d$.
- Jukes-Cantor: only one parameter, the rate of substitution, $\mu$, which is the same for all transitions. The JC69 model is a special case of the general time-reversible model, where the rate matrix is symmetric, $Q_{ij} = Q_{ji}$.
- K80 (Kimura): The model assumes base frequencies are equal for all characters, only one free parameter, the transition/transversion ratio, $\kappa$.
  $$
  Q = \beta
  \begin{bmatrix}
  -2 - \kappa & 1 & \kappa & 1 \\
  1 & -2 - \kappa & 1 & \kappa \\
  \kappa & 1 & -2 - \kappa & 1 \\
  1 & \kappa & 1 & -2 - \kappa
  \end{bmatrix}.
  $$

- F81 (Felsenstein 1981): Allows for unequal base frequencies ($\pi_A \neq \pi_C \neq \pi_G \neq \pi_T$), three free parameters, since the constraint that the four equilibrium frequencies must sum to 1 ($\pi_A + \pi_C + \pi_G + \pi_T = 1$).
  $$
  Q = \beta
  \begin{bmatrix}
  \pi_A - 1 & \pi_C & \pi_G & \pi_T \\
  \pi_A & \pi_C - 1 & \pi_G & \pi_T \\
  \pi_A & \pi_C & \pi_G - 1 & \pi_T \\
  \pi_A & \pi_C & \pi_G & \pi_T - 1
  \end{bmatrix}.
  $$

- HKY (Hasegawa, Kishino, and Yano 1985): combines the K80 and F81 models, allowing both unequal base frequencies and a transition/transversion ratio. The transition probabilities for the HKY model can be computed and have a closed-form solution, but it is rather long-winded.
  $$
  Q = \beta
  \begin{bmatrix}
  \cdot & \pi_C & \kappa \pi_G & \pi_T \\
  \pi_A & \cdot & \pi_G & \kappa \pi_T \\
  \kappa \pi_A & \pi_C & \cdot & \pi_T \\
  \pi_A & \kappa \pi_C & \pi_G & \cdot
  \end{bmatrix}.
  $$

- GTR (General Time Reversible): the most general time-reversible stationary CTMP for describing the
substitution process. The Q matrix is: 
  $$
  Q = \beta
  \begin{bmatrix}
  \cdot & a\pi_C & b\pi_G & c\pi_T \\
  a\pi_A & \cdot & d\pi_G & e\pi_T \\
  b\pi_A & d\pi_C & \cdot & \pi_T \\
  c\pi_A & e\pi_C & \pi_G & \cdot
  \end{bmatrix}.
  $$

## Codon models

- Codon models (Goldman and Yang 1994; Muse and Gaut 1994) have some key features:
  - a 61-codon state space (excluding the three stop codons);
  - a zero rate for substitutions that changed more than one nucleotide in a codon at any given instant (so each codon has **nine** immediate neighbors, minus any stop codons);
  - a synonymous/nonsynonymous bias parameter making synonymous mutations, that is mutations that do not change the protein that the codon codes for, more likely than nonsynonymous mutations.

- Given two codons $i = (i_1, i_2, i_3)$ and $j = (j_1, j_2, j_3)$, the Muse–Gaut-94 codon model has the following entries in the $Q$ matrix:

  $$
  q_{ij} =
  \begin{cases} 
  \beta \omega \pi_{jk} & \text{if nonsynonymous change at codon position } k \\
  \beta \pi_{jk} & \text{if synonymous change at codon position } k \\
  0 & \text{if codons } i \text{ and } j \text{ differ at more than one position}
  \end{cases}
  $$

  In addition, the Goldman–Yang-94 model includes a transition/transversion bias giving rise to entries in $Q$ as follows:

  $$
  q_{ij} =
  \begin{cases} 
  \beta \kappa \omega \pi_j & \text{if nonsynonymous transition} \\
  \beta \omega \pi_j & \text{if nonsynonymous transversion} \\
  \beta \kappa \pi_j & \text{if synonymous transition} \\
  \beta \pi_j & \text{if synonymous transversion} \\
  0 & \text{if codons differ at more than one position}
  \end{cases}
  $$

## Felsenstein’slikelihood

